src/components/inputs/Select/TextInput.js

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import TextField from '@material-ui/core/TextField';
+import { CircularProgress } from '@material-ui/core';
+
+import { useStyles } from '../styles';
+
+const TextInput = ({
+    params,
+    extraProps,
+    disabled,
+    label,
+    required,
+    onBlur,
+    errors,
+    helperText,
+    loading,
+}) => {
+    const classes = useStyles();
+    const paramsCopy = {
+        ...params,
+    };
+    let inputExtraProps = {};
+    if (extraProps.renderOption && params.inputProps.value) {
+        inputExtraProps = {
```

Comment:
if there's no other use of `extraProps` I would rename it since it seems we just use `extraProps.renderOption` to render the `startAdornment`.

`renderStartAdornment`?